pkg/migration-tool/utils/utils.go

```diff
@@ -106,7 +106,8 @@ func HashLabels(lset prompb.Labels) uint64 {
 		b = append(b, seps[0])
 	}
 	sum := xxhash.Sum64(b)
-	labelsCache.Insert(lset.String(), sum)
+	str := lset.String()
+	labelsCache.Insert(str, sum, uint64(len(str)+8+8))
```

Comment:
Is the best way to know the memory of string is by length? (I recall unsafe for a string always gave me 16 bits)

https://play.golang.com/p/_2_g5UTy2pT

IMO, we should do something like,
```suggestion
	labelsCache.Insert(str, sum, uint64(len([]byte(str))+8+8))
```
is more acurate.